demos/demo1-chat-heads/Demo.jsx

```diff
@@ -1,59 +1,222 @@
-import React from 'react';
-import {StaggeredMotion, spring, presets} from '../../src/react-motion';
-import range from 'lodash.range';
+import React from "react";
+import {
+  StaggeredMotion,
+  spring,
+  Motion,
+  presets
+} from "../../src/react-motion";
+import range from "lodash.range";
+import Slider from "./Slider";
 
-export default class Demo extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {x: 250, y: 300};
-  };
+const options = [
+  {
+    description: "Constant",
+    value: "constant"
+  },
+  {
+    description: "Linear",
+    value: "linear"
+  },
+  {
+    description: "Inverse Linear",
+    value: "invertedLinear"
+  }
+];
+
+const Demo = React.createClass({
+  getInitialState() {
+    return {
+      x: 250,
+      y: 300,
+      stiffness: 100,
+      damping: 17,
+      precision: 0.01,
+      stiffnessBehaviour: "constant",
+      dampingBehaviour: "constant",
+      headerHeight: 0
+    };
+  },
 
   componentDidMount() {
-    window.addEventListener('mousemove', this.handleMouseMove);
-    window.addEventListener('touchmove', this.handleTouchMove);
-  };
+    window.addEventListener("mousemove", this.handleMouseMove);
+    window.addEventListener("touchmove", this.handleTouchMove);
+    this.setState({ headerHeight: this.divRef.clientHeight });
+  },
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (this.state.x !== nextState.x) {
```

Comment:
Consider putting all of `if` clauses that evaluate under `true` into a single `if` clause with an `OR` operator, thus, reducing the amount of code and repetition. Thoughts?

Reply:
Good idea. Perhaps even better would be to use PureComponent since it does a shallow comparison for you, removing the need to explicitly check if state or props have changed?

Reply:
@therewillbecode that's a good idea. I'd love to see that implemented.